File: TestDoubles/Java/MockSampleTest.java
import org.junit.*;
import static org.junit.Assert.*;
import static org.mockito.Mockito.*;
import static org.hamcrest.CoreMatchers.*;

public class MockSampleTest 
{
    @Test
    public void test_with_mock_object() 
    {
        HikerHelper mock = mock(HikerHelper.class);
        Hiker douglas = new Hiker(mock);

        when(mock.multiplier()).thenReturn(7);

        assertThat(douglas.answerFor(6), is(42));
    }
}

File: TestDoubles/C#/MockSampleTest.cs
using System;
using Moq;
using NUnit.Framework;

[TestFixture]
public class MockSampleTest
{
    [Test]
    public void test_with_mock_object() 
    {
        var mock = new Mock<HikerHelper>();
        mock.Setup((m => m.GetMultiplier())).Returns(7);

        Hiker douglas = new Hiker(mock.Object);

        Assert.That(douglas.AnswerFor(6), Is.EqualTo(42));
    }
}
